### PR TITLE
Add support for ILanguageClient new methods

### PR DESCRIPTION
**Pull request by rchiodo**
ILanguageClient is changing (well because it can) and we need to change with it.

Fixes: #6609 

Tested with bits here:
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=4998492&view=ms.vss-build-web.run-extensions-tab

However we need to wait for that submission to go into main (or put this submission in at the same time) before we finish the PR into VS main.

Miguel should provide a branch we can insert into after Wednesday.

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/DidChangeWatchedFilesRegistrationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/DidChangeWatchedFilesRegistrationOptions.cs:1]: license header?

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/DidChangeWatchedFilesRegistrationOptions.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+using System.Collections.Generic;
+using System.Linq;
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/DidChangeWatchedFilesRegistrationOptions.cs:3]: `using System.Linq;` actually used?

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/DidChangeWatchedFilesRegistrationOptions.cs (hunk 1 above)
Comment (rchiodo) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/DidChangeWatchedFilesRegistrationOptions.cs:1]: Good point. Will fix

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/DidChangeWatchedFilesRegistrationOptions.cs (hunk 2 above)
Comment (rchiodo) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/DidChangeWatchedFilesRegistrationOptions.cs]: Should be fixed now

**Review by bschnurr — APPROVED** [review]

**Pull request merged**